server/route.go

```diff
@@ -423,7 +423,10 @@ func (s *Server) routeSidQueueSubscriber(rsid []byte) (*subscription, bool) {
 	}
 	sid := matches[RSID_SID_INDEX]
 
-	if sub, ok := client.subs[string(sid)]; ok {
+	client.mu.Lock()
```

Comment:
Actually, we grab the client's lock just before this call, wonder if we could not consolidate. Unless we explicitly want finding the sid out of matches to be done outside of a lock.